src/main/java/com/iabtcf/encoder/field/BooleanEncoder.java

```diff
@@ -11,10 +15,25 @@ private BooleanEncoder() {
     }
 
     private static final BooleanEncoder instance = new BooleanEncoder();
+
     public static BooleanEncoder getInstance() {
         return instance;
     }
-    public final Boolean decode(String value){
-        return value.equals("1");
+
+    private static final Logger logger = LogManager.getLogger(IntEncoder.class);
+
+    public final Boolean decode(String value) {
+        try {
+            if (value.equals(TRUE)) {
```

Comment:
The idea is to identify the difference between "0", "1", null and invalid values ("ABC", "$ 231", "", among others). Because it is important to detect whether the string is invalid.

Reply:
If we've made it all the way down to this decoder, we should know that it was invalid because the base64Url decoder checks to make sure it can be decoded into a value:
https://github.com/InteractiveAdvertisingBureau/iabtcf-java/blob/master/src/main/java/com/iabtcf/encoder/Base64Url.java#L35

Reply:
If anything, this should use a regex match for a "1" or a "0" and throw an error if it is not either of those because it's an error case that should blow up the whole decode.

Reply:
If I no longer use the null as a return, the following tests will no longer pass:
- **BooleanEncoderTest**
whenValueIsEmptyShouldReturnNull
whenValueIsInvalidShouldReturnNull
whenValueIsNullShouldReturnNull

- **FixedVectorEncoderTest**
whenValueIsInvalidThenShouldReturnNull

So do I delete them?

Reply:
I really believe we should be throwing an error instead of returning `null` because null is really a terminal case.  The values should have been checked further upstream before an encoding was to occur.

Reply:
Otherwise you'll have to check for `null` and do something with it, which I believe will end up being halting execution.

Reply:
You could use a Precondition check (guava Library for example) for the parameter at the beginning of the method (which will actually throw a NullPointerException in case the parameter is null). The calling code on the other hand has to make sure no nulls are passed to the method.
Your tests could then look like that (pseudo code):
`
@Test(expected = NullPointerException.class)
public void testNull() {
    myBooleanDecoder.decode(null);
}
`

Reply:
I will change this.